server/fleet/hosts.go

```diff
@@ -155,7 +155,7 @@ type HostDetail struct {
 	// Packs is the list of packs the host is a member of.
 	Packs []*Pack `json:"packs"`
 	// Policies is the list of policies and whether it passes for the host
-	Policies []*HostPolicy `json:"policies"`
+	Policies []*HostPolicy `json:"policies,omitempty"`
```

Comment:
We have to be careful that the slice is not nil if `IncludePolicies` is true. Currently, `*Datastore.ListPoliciesForHost` returns a nil slice if there are no policies and will get omitted.

Reply:
good catch, thanks! I'm wondering if for the sake of simplicity, removing `omitempty` and getting `"policies": null` when policies are not included is OK

Reply:
(this implementation doesn't cause errors because the UI does extra checks, but I agree is not ideal)

Reply:
May want to confirm with @lukeheath. At least when I added the cve scores to the api, we actually omitted the scores in the free version (as opposed to returning nulls). It required some double pointer magic to achieve. I don't think it would be too much work

```
	if opts.IncludePolicies {
		policies, err = svc.ds.ListPoliciesForHost(ctx, host)
		if err != nil {
			return nil, ctxerr.Wrap(ctx, err, "get policies for host")
		}
		if policies == nil {
			// initialize empty slice so it doesn't get omitted
			policies = make([]*fleet.HostPolicy, 0)
		}
	}
```

Reply:
thanks! 💜

Reply:
taking a look at this with fresh eyes, from the docs:

> The "omitempty" option specifies that the field should be omitted from the encoding if the field has an empty value, defined as false, 0, a nil pointer, a nil interface value, and any empty array, slice, map, or string. 

so we end up in the same situation even if we return a non-nil, empty slice. Another option could be to implement our own marshaller for `HostDetail` 🤔

Reply:
Ah, you are right. Either a pointer to a slice or a custom marshaler would work.